### PR TITLE
Pre-commit.con hook support

### PR DESCRIPTION
**Pull request by joelhess**
Please answer the following questions and leave the below in as part of your PR.

- [x] I have read the [developer documentation](https://github.com/clj-kondo/clj-kondo/blob/master/doc/dev.md).

- [x] This PR corresponds to an [issue with a clear problem statement](https://github.com/clj-kondo/clj-kondo/issues/1705).

- [ ] This PR contains a [test](https://github.com/clj-kondo/clj-kondo/blob/master/doc/dev.md#tests) to prevent against future regressions

- [x] I have updated the [CHANGELOG.md](https://github.com/clj-kondo/clj-kondo/blob/master/CHANGELOG.md) file with a description of the addressed issue.

Add .pre-commit-hook.yaml file to support integration with pre-commit tooling. Check out [pre-commit.com](https://pre-commit.com/) for additional details

### CODE REVIEW

File: .pre-commit-hooks.yaml [hunk 1]
```diff
@@ -0,0 +1,6 @@
+-   id: clj-kondo
+    name: clj-kondo
+    description: '`clj-kondo` is a command-line utility for enforcing style consistency across Clojure projects.'
+    entry: clj-kondo --lint
+    language: system
+    types: [clojure]
```
Comment (borkdude) [inline .pre-commit-hooks.yaml:6]: Does this also cover clojurescript?

**Comment by borkdude** [comment]
Thanks!

File: .pre-commit-hooks.yaml (hunk 1 above)
Comment (joelhess) [reply .pre-commit-hooks.yaml:6]: Yes. It'll cover .Clj .cljs and .edn files.

Comment (borkdude) [reply .pre-commit-hooks.yaml:6]: Thanks!

**Pull request merged**